openapi_python_client/templates/property_templates/model_property.pyi

```diff
@@ -25,19 +25,25 @@ if {% if property.nullable %}_{{ property.python_name }} is not None{% endif %}{
 {% macro check_type_for_construct(source) %}isinstance({{ source }}, dict){% endmacro %}
 
 {% macro transform(property, source, destination, declare_type=True, query_parameter=False) %}
+{{ transform_extended(property, source, destination, declare_type=declare_type, query_parameter=query_parameter) }}
+{% endmacro %}
+
+{% macro transform_extended(property, source, destination, declare_type=True, query_parameter=False, skip_read_only_expr="") %}
```

Comment:
The `transform` macro is what the generator always calls to generate the serialization logic for whatever type it is. I added this `transform_extended` version because if I had just added an optional parameter to `transform`, it wouldn't match the signature of the `transform` macros that appear in other property-type templates.

The semantics here are, if we're being called from the `endpoint_macros` template where it's constructing a request, and the class we're serializing is the request body, then `skip_read_only_expr` will be `"True"`. But, we could instead be serializing a nested object inside some other class, which might or might not be in a request body. In that case, we get called from the `model.pyi` template, where there is a local variable/parameter that's keeping track of whether we are in skip-read-only-properties mode... so `skip_read_only_expr` will be set to the name of that parameter.